File: create_admin.js
const { Client } = require('pg');
const crypto = require('crypto');

// pbkdf2_sha256 hash compativel com passlib
function hashPassword(password) {
    const salt = crypto.randomBytes(16).toString('base64');
    const iterations = 600000;
    const keylen = 32;
    const hash = crypto.pbkdf2Sync(password, salt, iterations, keylen, 'sha256').toString('base64');
    return `$pbkdf2-sha256$i=${iterations},l=${keylen}$${salt}$${hash}`;
}

const DATABASE_URL = 'postgresql://postgres:USWVMOOGZXpocoBbNvRRKzxnjzumBNCw@shortline.proxy.rlwy.net:44238/railway';
const EMAIL = 'admin@sistema.com';
const SENHA = 'admin123';
const NOME = 'Administrador';

async function createAdmin() {
    const client = new Client({ connectionString: DATABASE_URL });

    try {
        await client.connect();
        console.log('Conectado ao banco Railway...');

        const hashed = hashPassword(SENHA);

        // Verificar se usuario ja existe
        const existing = await client.query('SELECT id, email, role FROM users WHERE email = $1', [EMAIL]);

        if (existing.rows.length > 0) {
            await client.query(
                'UPDATE users SET role = $1, senha = $2 WHERE email = $3',
                ['ADMIN', hashed, EMAIL]
            );
            console.log('✅ Usuario existente atualizado para ADMIN: ' + EMAIL);
        } else {
            const result = await client.query(
                'INSERT INTO users (nome, email, senha, role, ativo) VALUES ($1, $2, $3, $4, $5) RETURNING id',
                [NOME, EMAIL, hashed, 'ADMIN', true]
            );
            console.log('✅ Usuario administrador criado!');
            console.log('   ID: ' + result.rows[0].id);
            console.log('   Email: ' + EMAIL);
            console.log('   Senha: ' + SENHA);
            console.log('   Role: ADMIN');
        }
    } catch (err) {
        console.error('Erro:', err.message);
        throw err;
    } finally {
        await client.end();
    }
}

createAdmin().catch(console.error);

File: create_admin.py
import sys
import os

# Instalar dependencias necessarias se nao tiver
try:
    from passlib.context import CryptContext
    import psycopg2
except ImportError:
    os.system(f"{sys.executable} -m pip install passlib psycopg2-binary --quiet")
    from passlib.context import CryptContext
    import psycopg2

DATABASE_URL = "postgresql://postgres:USWVMOOGZXpocoBbNvRRKzxnjzumBNCw@shortline.proxy.rlwy.net:44238/railway"

EMAIL = "admin@sistema.com"
SENHA = "admin123"
NOME = "Administrador"

pwd_context = CryptContext(schemes=["pbkdf2_sha256"], deprecated="auto")
hashed = pwd_context.hash(SENHA)

conn = psycopg2.connect(DATABASE_URL)
cur = conn.cursor()

# Verificar se usuario ja existe
cur.execute("SELECT id, email, role FROM users WHERE email = %s", (EMAIL,))
existing = cur.fetchone()

if existing:
    # Atualizar para ADMIN se ja existe
    cur.execute("UPDATE users SET role = 'ADMIN', senha = %s WHERE email = %s", (hashed, EMAIL))
    conn.commit()
    print(f"✅ Usuario existente atualizado para ADMIN: {EMAIL}")
else:
    # Criar novo usuario admin
    cur.execute(
        "INSERT INTO users (nome, email, senha, role, ativo) VALUES (%s, %s, %s, %s, %s) RETURNING id",
        (NOME, EMAIL, hashed, "ADMIN", True)
    )
    user_id = cur.fetchone()[0]
    conn.commit()
    print(f"✅ Usuario administrador criado com sucesso!")
    print(f"   ID: {user_id}")
    print(f"   Email: {EMAIL}")
    print(f"   Senha: {SENHA}")
    print(f"   Role: ADMIN")

cur.close()
conn.close()
print("\n🎉 Pronto! Faça login com as credenciais acima.")
